common/types/types.ts

```diff
@@ -18,7 +18,7 @@ export type Language = {
   label: string;
   code: LanguageCode;
   rtl: boolean;
-  isFavorite: boolean;
+  isFavorite?: boolean;
```

Comment:
Why do we need `isFavorite` to be undefined-able? Is it because it doesn't always make sense to set the property at all? Could we have a `UserLanguage` type that extends `Language` and adds `isFavorite: boolean`?